src/main/java/com/minecraftabnormals/buzzier_bees/core/other/BBCompat.java

```diff
@@ -61,6 +120,33 @@ public static void registerDispenserBehaviors() {
 		DispenserBlock.registerDispenseBehavior(BBItems.BOTTLE_OF_BEE.get(), new BeeBottleDispenseBehavior());
 		DispenserBlock.registerDispenseBehavior(BBItems.BOTTLE_OF_SILVERFISH.get(), new BugBottleDispenseBehavior());
 		DispenserBlock.registerDispenseBehavior(BBItems.BOTTLE_OF_ENDERMITE.get(), new BugBottleDispenseBehavior());
+
+		DataUtil.registerAlternativeDispenseBehavior(Items.GLASS_BOTTLE, (source, stack) -> !BlockUtil.getEntitiesAtOffsetPos(source, CreatureEntity.class, e -> ENTITY_TYPE_TO_BOTTLE_MAP.containsKey(e.getType())).isEmpty(), new DefaultDispenseItemBehavior() {
+			@Override
+			protected ItemStack dispenseStack(IBlockSource source, ItemStack stack) {
+				Entity entity = BlockUtil.getEntitiesAtOffsetPos(source, Entity.class, e -> ENTITY_TYPE_TO_BOTTLE_MAP.containsKey(e.getType())).get(0);
+				ItemStack bottled = ENTITY_TYPE_TO_BOTTLE_MAP.get(entity.getType()).apply(entity);
+				stack.shrink(1);
+				if (stack.isEmpty()) {
+					stack = bottled.copy();
+				} else if (source.<DispenserTileEntity>getBlockTileEntity().addItemStack(bottled.copy()) < 0) {
+					new DefaultDispenseItemBehavior().dispense(source, bottled.copy());
+				}
+				source.getWorld().playSound(null, source.getBlockPos(), SoundEvents.ITEM_BOTTLE_FILL_DRAGONBREATH, SoundCategory.NEUTRAL, 1.0F, 1.0F);
+				source.getWorld().removeEntity(entity);
+				return stack;
+			}
+		});
+		DataUtil.registerAlternativeDispenseBehavior(Items.FLINT_AND_STEEL, (source, stack) -> BlockUtil.getStateAtOffsetPos(source).getBlock() instanceof CandleBlock && !BlockUtil.getStateAtOffsetPos(source).get(BlockStateProperties.LIT), new DefaultDispenseItemBehavior() {
```

Comment:
`BlockUtil.getStateAtOffsetPos(source)` could be moved to a local variable.

Reply:
Added in latest commit